src/components/TrashCard/TrashCard.tsx

```diff
@@ -2,13 +2,61 @@ import { MdOutlineDelete, MdOutlineRestoreFromTrash } from "react-icons/md";
 import { useState } from "react";
 
 import { TrashCardProps } from "../component.types";
+import { useData } from "../../context";
+import { addNote } from "../../utils/server-request";
+import toast from "react-hot-toast";
 
-export const TrashCard = ({ title, noteText, id }: TrashCardProps) => {
+export const TrashCard = ({ title, noteText, id, bgColor }: TrashCardProps) => {
   const [showActionBtn, setShowActionBtn] = useState(false);
+  const [isInProcess, setIsInProcess] = useState(false);
+
+  const { notes, setNotes } = useData();
+  const uid = localStorage.getItem("userId") || "";
+
+  const handleDeleteNote = async (id: string) => {
+    setIsInProcess(true);
+    try {
+      const updatedNoteList = notes.filter((note) => note.id !== id);
+
+      await addNote(updatedNoteList, uid);
+      setNotes(updatedNoteList);
+      toast.success("Note deleted permanently");
+      setIsInProcess(false);
+    } catch (error) {
+      console.log(error);
+      toast.error("Please try again later.");
+      setIsInProcess(false);
```

Comment:
suggestion:  you can write "setIsInProcess" in finally block.

Reply:
Noted, I will change in later PR